pkg/processor/eventprocessor/statistics.go

```diff
@@ -18,34 +18,6 @@ package eventprocessor
 
 import "sync/atomic"
 
-type Statistics struct {
-	EventsHandledSuccess uint64
-	EventsHandledError   uint64
-}
-
-// AllocatorStatistics is not a safe statistics object and should be used only to copy safe object to it and return to outside
-// so it can later be taken by metric gatherers
-type AllocatorStatistics struct {
-	AllocationCount                       uint64
-	AllocationSuccessImmediateTotal       uint64
-	AllocationSuccessAfterWaitTotal       uint64
-	AllocationTimeoutTotal                uint64
-	AllocationWaitDurationMilliSecondsSum uint64
-	AllocationObjectsAvailablePercentage  uint64
-}
-
-func (s *AllocatorStatistics) DiffFrom(prev *AllocatorStatistics) AllocatorStatistics {
-
-	return AllocatorStatistics{
-		AllocationCount:                       s.AllocationCount - prev.AllocationCount,
-		AllocationSuccessImmediateTotal:       s.AllocationSuccessImmediateTotal - prev.AllocationSuccessImmediateTotal,
-		AllocationSuccessAfterWaitTotal:       s.AllocationSuccessAfterWaitTotal - prev.AllocationSuccessAfterWaitTotal,
-		AllocationTimeoutTotal:                s.AllocationTimeoutTotal - prev.AllocationTimeoutTotal,
-		AllocationWaitDurationMilliSecondsSum: s.AllocationWaitDurationMilliSecondsSum - prev.AllocationWaitDurationMilliSecondsSum,
-		AllocationObjectsAvailablePercentage:  s.AllocationObjectsAvailablePercentage - prev.AllocationObjectsAvailablePercentage,
-	}
-}
-
 // safeAllocatorStatistics is a safe statistics object, for outside usages use AllocatorStatistics
 type safeAllocatorStatistics struct {
```

Comment:
move it to `metrics/types.go` too, and get rid of this file

Reply:
@TomerShor the whole point is to have it private, event processor-specific, so I'd leave it here

Reply:
the instance of this type can still be private, even if the type itself is not

Reply:
I understand, but it basically duplicates the `AllocatorStatistics` struct with atomic values.
To me it makes sense to have them in the same package (don't really care about files).

Reply:
Yeah it duplicates, but we aren't gonna use it outside of eventprocessor, so I'd leave it here